FrameworkSystem/Service/SystemAdministratorHandler.py

```diff
@@ -54,7 +55,12 @@ def initializeHandler( cls, serviceInfo ):
       global gMonitoringReporter
       gMonitoringReporter = MonitoringReporter( monitoringType = "ComponentMonitoring" )
       gThreadScheduler.addPeriodicTask( 120, cls.__storeProfiling )
-      
+    
+    keepSoftwareVersions = cls.srv_getCSOption( 'KeepSoftwareVersions', 0 )
+    if keepSoftwareVersions:
+      gLogger.info( "The last %s software version will be kept and the rest will be deleted!" % keepSoftwareVersions )
+      gThreadScheduler.addPeriodicTask( 120, cls.__deleteOldSoftware, ( keepSoftwareVersions, ), executions = 2 ) #it is enough to try 2 times
```

Comment:
I do not think so. It is enough to clean the directory after the release. I want to implement, the following ( @chaen this is the answer for your question):
-after when we update the machine, we clean the directory (and we try two times).  I can not trigger the deletion from the update method, because the following reasons:
  1.) We slow down the update time
  2.) after we update the machine, we immediately restart the system administrator. So, If we trigger a background process, it will killed as the service is restarted. 
I would say It is enough to only try to clean after the update of the machine taking account we always restart the SystemAdministtartor after update. I do not see the need of a periodic task.